src/common/transformations/src/transformations/common_optimizations/reverse_shape_and_type_infer.cpp

```diff
@@ -99,9 +100,10 @@ bool ov::pass::ReverseShapeAndTypeInfer::run_on_model(const std::shared_ptr<ov::
         if (const auto& param = ov::as_type_ptr<ov::op::v0::Parameter>(op)) {
             if (param->get_partial_shape().compatible(output_shape)) {
                 auto shape = param->get_partial_shape();
-                PartialShape::merge_into(shape, output_shape);
-                param->set_partial_shape(shape);
-                is_changed = true;
+                if (PartialShape::merge_into(shape, output_shape)) {
```

Comment:
what should we do if `merge_into` returned false? just ignore it?
as far as I remember, `merge_into` might modify the `shape` argument
@mvafin what do you think?

Reply:
It will fail only when shapes are incompatible? Or when shape it is merging to is more restrictive?

If it fails for restrictive, but compatible case I think we can ignore it, because it might be user specified shape
Otherwise I would treat it as error